packages/main/src/components/ActionSheet/index.tsx

```diff
@@ -84,36 +161,76 @@ const ActionSheet: FC<ActionSheetPropTypes> = forwardRef(
       'onBeforeClose',
       'onBeforeOpen'
     ]);
+    const handleAfterOpen = useCallback(
+      (e) => {
+        if (isPhone()) {
+          actionBtnsRef.current.querySelector(`[data-action-btn-index="${focusedItem}"]`).focus();
+        }
+        if (typeof onAfterOpen === 'function') {
+          onAfterOpen(e);
+        }
+      },
+      [onAfterOpen, actionBtnsRef.current, focusedItem]
+    );
+    const displayHeader = alwaysShowHeader || isPhone();
 
     return createPortal(
       <ResponsivePopover
-        ref={popoverRef}
+        aria-modal
+        role="dialog"
         style={style}
         slot={slot}
-        className={actionSheetClasses.className}
         allowTargetOverlap={allowTargetOverlap}
-        headerText={headerText}
+        headerText={displayHeader ? headerText : undefined}
         horizontalAlign={horizontalAlign}
         initialFocus={initialFocus}
         modal={modal}
         noArrow={noArrow}
         placementType={placementType}
         verticalAlign={verticalAlign}
         footer={footer}
-        header={header}
+        header={displayHeader ? header : undefined}
         onAfterClose={onAfterClose}
-        onAfterOpen={onAfterOpen}
         onBeforeClose={onBeforeClose}
         onBeforeOpen={onBeforeOpen}
         {...passThroughProps}
+        onAfterOpen={handleAfterOpen}
+        ref={popoverRef}
+        className={actionSheetClasses.className}
+        data-actionsheet
       >
-        {Children.map(children, renderActionSheetButton)}
+        <div
+          className={isPhone() ? classes.contentMobile : undefined}
+          role="presentation"
+          aria-label={i18nBundle.getText(AVAILABLE_ACTIONS)}
+          onKeyDown={handleKeyDown}
+          ref={actionBtnsRef}
+        >
+          {Children.map(children, renderActionSheetButton)}
```

Comment:
```suggestion
          {childrenToRender.map(renderActionSheetButton)}
```
for conditional rendering support